api/v1alpha1/tls_types.go

```diff
@@ -0,0 +1,98 @@
+// Copyright Envoy Gateway Authors
+// SPDX-License-Identifier: Apache-2.0
+// The full text of the Apache license is available in the LICENSE file at
+// the root of the repo.
+
+package v1alpha1
+
+// +kubebuilder:validation:XValidation:rule="self.version.min == 'TLSv1_3' ? !has(self.ciphers) : true", message="setting ciphers has no effect if the minimum possible TLS version is 1.3"
+type TLSSettings struct {
+
+	// Version details the minimum/maximum TLS protocol version that
+	// should be supported by this listener.
+	// +optional
+	Version *TLSVersions `json:"version,omitempty"`
+
+	// CipherSuites specifies the set of cipher suites supported when
+	// negotiating TLS 1.0 - 1.2. This setting has no effect for TLS 1.3.
+	//
+	// In non-FIPS Envoy Proxy builds, the default cipher list is:
+	// - [ECDHE-ECDSA-AES128-GCM-SHA256|ECDHE-ECDSA-CHACHA20-POLY1305]
+	// - [ECDHE-RSA-AES128-GCM-SHA256|ECDHE-RSA-CHACHA20-POLY1305]
+	// - ECDHE-ECDSA-AES256-GCM-SHA384
+	// - ECDHE-RSA-AES256-GCM-SHA384
+	//
+	// In builds using BoringSSL FIPS the default cipher list is:
+	// - ECDHE-ECDSA-AES128-GCM-SHA256
+	// - ECDHE-RSA-AES128-GCM-SHA256
+	// - ECDHE-ECDSA-AES256-GCM-SHA384
+	// - ECDHE-RSA-AES256-GCM-SHA384
+	//
+	// +optional
+	CipherSuites []string `json:"ciphers,omitempty"`
```

Comment:
should this be made into an enum for better validation ?

Reply:
It's not possible to make it into an enum, because the values in here are not always singular values. 

See the [cipher list documentation](https://commondatastorage.googleapis.com/chromium-boringssl-docs/ssl.h.html#Cipher-suite-configuration) from Envoy Proxy's underlying library.

The values in this array are expressions, allowing things like `[ECDHE-ECDSA-AES128-GCM-SHA256|ECDHE-ECDSA-CHACHA20-POLY1305]`, which is an expression that says "either the first cipher or the second".

Additional possible values are to use `!` or '+' or '-' as the prefix for a cipher name. This cannot be expressed as an enum in Envoy Gateway.

Reply:
cool, lets keep it as is